rapid7_insightvm/komand_rapid7_insightvm/util/resource_requests.py

```diff
@@ -0,0 +1,203 @@
+        requests.ConnectionError: 'Unable to connect to IVM console.'
+                                  'If this issue persists contact support for assistance.',
+        requests.Timeout: _ENSURECONNECTIVITY,
+        requests.ConnectTimeout: _ENSURECONNECTIVITY,
+        requests.ReadTimeout: _ENSURECONNECTIVITY,
+        requests.TooManyRedirects: _ENSURECONNECTIVITY
+    }
+
+    # For request exceptions not in REQUEST_EXCEPTIONS
+    _UNHANDLED_EXCEPTION = 'Contact support for assistance'
+
+    def __init__(self, session, logger):
+        """
+        Creates a new instance of ResourceHelper
+        :param session: Session object available to Komand actions/triggers, usually self.connection.session
+        :param logger: Logger object available to Komand actions/triggers, usually self.logger
+        :return: ResourceHelper object
+        """
+        self.logger = logger
+        self.session = session
+        self.session.headers.update(self._HEADERS)
+
+        # Suppress insecure request messages
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    def resource_request(self, endpoint: str, method: str = 'get', params: Collection = None, payload: dict = None,
+                         json_response: bool = True) -> dict:
+        """
+        Sends a request to APIv3 with the provided endpoint and optional method/payload
+        :param endpoint: Endpoint for the APIv3 call defined in endpoints.py
+        :param method: HTTP method for the API request
+        :param params: URL parameters to append to the request
+        :param payload: JSON body for the API request if required
+        :param json_response: Boolean to return raw response
+        :return: Dict containing the JSON response body
+        """
+
+        request_method = getattr(self.session, method.lower())
+        if not payload:
+            payload = dict()
+        if not params:
+            params = dict()
+        if isinstance(params, list):
+            parameters = RequestParams.from_tuples(params)
+        else:
+            parameters = RequestParams.from_dict(params)
+        if 'rawbody' in payload.keys():
+            payload = payload['rawbody']
+
+        extras = {"json": payload, "params": parameters.params}
+        try:
+            response = request_method(url=endpoint, verify=False, **extras)
+        except requests.RequestException as e:
+            assistance = self._REQUEST_EXCEPTIONS.get(type(e), self._UNHANDLED_EXCEPTION)
+            raise PluginException(cause=e,
+                                  assistance=assistance)
+
+        resource_request_status_code_check(response.text, response.status_code)
+
+        if json_response:
+            try:
+                resource = response.json()
+            except json.decoder.JSONDecodeError as e:
+                raise PluginException(cause=f'Error: Received an unexpected response from InsightVM '
```

Comment:
The word 'Error:' is redundant because the SDK prefixes the line with a failure message. We could just use the `preset` JSON message included in the SDK here.

```suggestion
                raise PluginException(cause=f'Received an unexpected response from InsightVM '
```